controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -384,6 +386,52 @@ func (in *KubeadmControlPlane) validateEtcd(prev *KubeadmControlPlane) (allErrs
 	return allErrs
 }
 
+func (in *KubeadmControlPlane) validateVersion(previousVersion string) (allErrs field.ErrorList) {
+	fromVersion, err := util.ParseMajorMinorPatch(previousVersion)
+	if err != nil {
+		allErrs = append(allErrs,
+			field.InternalError(
+				field.NewPath("spec", "version"),
+				errors.Wrapf(err, "failed to parse current kubeadmcontrolplane version: %s", previousVersion),
+				),
+		)
+		return allErrs
+	}
+
+	toVersion, err := util.ParseMajorMinorPatch(in.Spec.Version)
+	if err != nil {
+		allErrs = append(allErrs,
+			field.InternalError(
+				field.NewPath("spec", "version"),
+				errors.Wrapf(err, "failed to parse current kubeadmcontrolplane version: %s", previousVersion),
+			),
+		)
+		return allErrs
+	}
+
+	if toVersion.LT(fromVersion) {
+		allErrs = append(allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "version"),
+				fmt.Sprintf("cannot downgrade control plane from %s to %s", previousVersion, in.Spec.Version),
+			),
+		)
```

Comment:
Should we allow rollback though? This seems it might impact the ability to stop and rollback an upgrade.

I'd leave this out for now and only validate the upgrade path, where we need to check that you can only upgrade one minor version

cc @detiber

Reply:
Alright, I will remove the downgrade part for now.